packages/core/strapi/lib/services/event-hub.js

```diff
@@ -1,16 +1,30 @@
+'use strict';
+
 /**
  * The event hub is Strapi's event control center.
  */
+module.exports = function createEventHub() {
+  // Store of subscribers that will be called when an event is emitted
+  const subscribers = [];
 
-'use strict';
+  return {
```

Comment:
to keep the on implementation we could register a default subscriber and keep a map of event to callback subs (we do sth similar in the db lifecycles to support targeting specific eventNames or have a generic subscriber)